src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -176,6 +180,24 @@ default boolean isEmpty() {
     byte[] getBases();
 
     /**
+     * @return The number of bases in the read sequence.
+     * This default implementation calls getBases().length
+     * Subclasses may override.
+     */
+    default int getBaseCount(){
+        return getBases().length;
+    }
+
+    /**
+     * @return The base at index i.
+     * This default implementation returns getBases()[i].
+     * Subclasses may override.
+     */
+    default byte getBase(final int i){
```

Comment:
I think there's no need to specify that subclasses may override a default method.  It might say WHY a subclass would override though.  I think it's reasonable to specify that it MUST be consistent with getBases()[i] as well.